### PR TITLE
Tax categories and shipping categories are now archived instead of deleted

### PR DESCRIPTION
**Pull request by pdaleramirez**
Changed (#1815, #2431)

### CODE REVIEW

File: src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * m201207_065953_add_dataDeleted_column_archive_and_tax_categories migration.
+ */
+class m201207_065953_add_dataDeleted_column_archive_and_tax_categories extends Migration
+{
+    /**
+     * @inheritdoc
+     */
+    public function safeUp()
+    {
+        if (!$this->db->columnExists('{{%commerce_shippingcategories}}', 'pdfLanguage')) {
```
Comment (lukeholder) [inline src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php:18]: I think the wrong column name is checked here.

File: src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            $this->addColumn('{{%commerce_shippingcategories}}', 'dateDeleted', $this->dateTime()->after('dateUpdated'));
+        }        
+        
+        if (!$this->db->columnExists('{{%commerce_taxcategories}}', 'pdfLanguage')) {
```
Comment (lukeholder) [inline src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php:22]: I think the wrong column name is checked here.

**Review by lukeholder — CHANGES_REQUESTED** [review]
Can we get some tests made for this feature? See also the other comments:

File: src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php (hunk 1 above)
Comment (pdaleramirez) [reply src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php]: I fixed this on latest commit

File: src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php (hunk 2 above)
Comment (pdaleramirez) [reply src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php]: I fixed this on latest commit

**Comment by pdaleramirez** [comment]
> Can we get some tests made for this feature? See also the other comments:

I didn't write any unit tests for this feature because I think It is better to write functional test for this feature.

File: src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php (hunk 1 above)
Comment (pdaleramirez) [reply src/migrations/m201207_065953_add_dataDeleted_column_archive_and_tax_categories.php]: > Can we get some tests made for this feature? See also the other comments:

I created the ShippingCategoryTest with fixture under the unit test folder

**Review by lukeholder — CHANGES_REQUESTED** [review]
Lets get this updated with the latest from 3.4.x and then request my review again @pdaleramirez

**Comment by pdaleramirez** [comment]
> Lets get this updated with the latest from 3.4.x and then request my review again @pdaleramirez

@lukeholder I've updated the branch to the latest develop branch. I've tested the unit test and all passed. It is ready to merged.

**Comment by pdaleramirez** [comment]
New PR for the feature here https://github.com/craftcms/commerce/pull/2912

**Pull request closed without merging**